src/frontend/src/lib/services/reward-code.services.ts

```diff
@@ -0,0 +1,63 @@
+import type { VipReward } from '$declarations/rewards/rewards.did';
+import {
+	claimVipReward as claimVipRewardApi,
+	getNewVipReward as getNewVipRewardApi,
+	getUserInfo as getUserInfoApi
+} from '$lib/api/reward.api';
+import { i18n } from '$lib/stores/i18n.store';
+import type { Identity } from '@dfinity/agent';
+import { fromNullable } from '@dfinity/utils';
+import { get } from 'svelte/store';
+
+export const isVipUser = async ({
```

Comment:
Are you planning to handle potential errors from the services on the UI side?

It makes sense to provide all the APIs, but should we perhaps provide the service wrappers only when necessary? I assume we’ll first review, for example, the usage of `isVipUser` before implementing `claimVipReward`, etc. I’m concerned about adding dead code or code that might require several iterations to refine, you know what I mean?